packages/cli/test/mocks/client.ts

```diff
@@ -0,0 +1,109 @@
+import chalk from 'chalk';
+import { createServer, Server } from 'http';
+import express, { Express, Router } from 'express';
+import listen from 'async-listen';
+import Client from '../../src/util/client';
+import { Output } from '../../src/util/output';
+
+// Disable colors in `chalk` so that tests don't need
+// to worry about ANSI codes
+chalk.level = 0;
+
+export type Scenario = Router;
+
+export class MockClient extends Client {
+  mockServer?: Server;
+  mockOutput: jest.Mock<void, Parameters<Output['print']>>;
+  private app: Express;
+  scenario: Scenario;
+
+  constructor() {
+    super({
+      argv: [],
+      // Gets populated in `startMockServer()`
+      apiUrl: '',
+      authConfig: {},
+      output: new Output(),
+      config: {},
+      localConfig: {},
+    });
+    this.mockOutput = jest.fn();
+
+    this.app = express();
+    this.app.use(express.json());
```

Comment:
Why use `express` here? We don't really use it anywhere else as far as I know?

Reply:
Mainly just for the decent router capabilities. Open to suggestions if you'd prefer a different framework.